docs/design/admission_control_resource_quota.md

```diff
@@ -54,109 +58,139 @@ const (
-namespace.  If acceptance of the resource would cause the total usage of a named resource to exceed its hard limit, the request is denied.
-
-The following resource limits are imposed as part of core Kubernetes at the namespace level:
-
-| ResourceName | Description |
+| Resource | Description |
 | ------------ | ----------- |
-| cpu | Total cpu usage |
-| memory | Total memory usage |
-| pods | Total number of pods  |
+| cpu | Total requested cpu usage |
+| memory | Total requested memory usage |
+| pods | Total number of active pods where phase is pending or active.  |
 | services | Total number of services |
 | replicationcontrollers | Total number of replication controllers |
 | resourcequotas | Total number of resource quotas |
+| secrets | Total number of secrets |
+| persistentvolumeclaims | Total number of persistent volume claims |
 
-Any resource that is not part of core Kubernetes must follow the resource naming convention prescribed by Kubernetes.
+If a third-party wants to track additional resources, it must follow the resource naming conventions prescribed
+by Kubernetes.  This means the resource must have a fully-qualified name (i.e. mycompany.org/shinynewresource)
 
-This means the resource must have a fully-qualified name (i.e. mycompany.org/shinynewresource)
+## Resource Requirements: Requests vs Limits
 
-If the incoming request does not cause the total usage to exceed any of the enumerated hard resource limits, the plug-in will post a
-**ResourceQuotaUsage** document to the server to atomically update the observed usage based on the previously read
-**ResourceQuota.ResourceVersion**.  This keeps incremental usage atomically consistent, but does introduce a bottleneck (intentionally)
-into the system.
+If a resource supports the ability to distinguish between a request and a limit for a resource,
+the quota tracking system will only cost the request value against the quota usage.  If a resource
+is tracked by quota, and no request value is provided, the associated entity is rejected as part of admission.
 
-To optimize system performance, it is encouraged that all resource quotas are tracked on the same **ResourceQuota** document.  As a result,
-its encouraged to actually impose a cap on the total number of individual quotas that are tracked in the **Namespace** to 1 by explicitly
-capping it in **ResourceQuota** document.
+For an example, consider the following scenarios relative to tracking quota on CPU:
 
-## kube-apiserver
+| Pod | Container | Request CPU | Limit CPU | Result |
+| --- | --------- | ----------- | --------- | ------ |
+| X | C1 | 100m | 500m | The quota usage is incremented 100m |
+| Y | C2 | 100m | none | The quota usage is incremented 100m |
+| Y | C2 | none | 500m | The quota usage is incremented 500m since request will default to limit |
+| Z | C3 | none | none | The pod is rejected since it does not enumerate a request. |
```

Comment:
Why does ResourceQuota reject pods that don't enumerate requests? Do we want users to explicitly specify a request of 0 to run memory best-effort containers?